astropy/table/table.py

```diff
@@ -817,29 +854,57 @@ def _init_from_list_of_dicts(self, data, names, dtype, n_cols, copy):
         # Note: if set(data[0].keys()) != names_from_data, this will give an
         # exception later, so NO need to catch here.
 
+        # Convert list of dict into dict of list (cols), keep track of missing
+        # indexes and put in MISSING placeholders in the `cols` lists.
         cols = {}
+        missing_indexes = defaultdict(list)
         for name in names_from_data:
             cols[name] = []
-            for i, row in enumerate(data):
+            for ii, row in enumerate(data):
                 try:
-                    cols[name].append(row[name])
+                    val = row[name]
                 except KeyError:
-                    raise ValueError(f'Row {i} has no value for column {name}')
-
+                    missing_indexes[name].append(ii)
+                    val = MISSING
+                cols[name].append(val)
+
+        # Fill the missing entries with first values
+        if missing_indexes:
+            first_vals = {}
+            for name in missing_indexes:
+                col = cols[name]
+                for val in col:
+                    if val is not MISSING:
+                        first_vals[name] = val
+                        break
+                for index in missing_indexes[name]:
```

Comment:
Actually, looking at your code below, can you not simply do `col[indexes] = first_vals[name]`?

Reply:
`col` is a list not np.array, so no joy there.

Reply:
But upon further inspection I realized `first_vals` doesn't need to be a dict, and from there found a more elegant way to get `first_val`.